### ISSUE TITLE
GBoard Preferences app keeps crashing

### ISSUE THREAD

**Issue by naXa777** [opengapps/aosp_build#116]
I have a Nexus 5 device and I've installed [Android 7.1.1 AOSP](https://forum.xda-developers.com/google-nexus-5/orig-development/wip-aosp-nougat-nexus-5-t3446875) and a customized [opengapps aroma](http://opengapps.org/) package via [TWRP](https://twrp.me/site/update/2016/04/05/twrp-3.0.2-0-released.html).
Everything plays well together except the GBoard not having Russian (Cyrillic) input. In order to fix this I headed to Settings > Languages & input > Virtual keyboard > GBoard and then I see a list of options (like Languages, Preferences, etc.). Tapping on any of them results in the GBoard application crash with the following stacktrace:

```
01-06 18:29:49.857	2256	5539	I	Icing	Usage reports 0 indexed 0 rejected 0 imm upload false
01-06 18:29:50.316	917	2394	I	ActivityManager	START u0 {act=android.intent.action.MAIN cmp=com.android.inputmethod.latin/com.google.android.apps.inputmethod.latin.preference.SettingsActivity (has extras)} from uid 10018 on display 0
01-06 18:29:50.372	7938	7938	D	AndroidRuntime	Shutting down VM
01-06 18:29:50.376	7938	7938	E	AndroidRuntime	FATAL EXCEPTION: main
01-06 18:29:50.376	7938	7938	E	AndroidRuntime	Process: com.android.inputmethod.latin, PID: 7938
01-06 18:29:50.376	7938	7938	E	AndroidRuntime	java.lang.RuntimeException: Unable to start activity ComponentInfo{com.android.inputmethod.latin/com.google.android.apps.inputmethod.latin.preference.SettingsActivity}: java.lang.RuntimeException: Preference xml file not specified
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.ActivityThread.-wrap12(ActivityThread.java)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.os.Handler.dispatchMessage(Handler.java:102)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.os.Looper.loop(Looper.java:154)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.ActivityThread.main(ActivityThread.java:6119)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at java.lang.reflect.Method.invoke(Native Method)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime	Caused by: java.lang.RuntimeException: Preference xml file not specified
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at com.google.android.apps.inputmethod.libs.framework.preference.CommonPreferenceFragment.onCreate(PG:26)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.Fragment.performCreate(Fragment.java:2336)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:949)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.BackStackRecord.setLastIn(BackStackRecord.java:860)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.BackStackRecord.calculateFragments(BackStackRecord.java:900)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.BackStackRecord.run(BackStackRecord.java:728)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1578)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.FragmentController.execPendingActions(FragmentController.java:371)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.Activity.performStart(Activity.java:6695)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2628)
01-06 18:29:50.376	7938	7938	E	AndroidRuntime		... 9 more
01-06 18:29:50.380	917	1133	W	ActivityManager	  Force finishing activity com.android.inputmethod.latin/com.google.android.apps.inputmethod.latin.preference.SettingsActivity
01-06 18:29:50.387	917	1133	W	ActivityManager	  Force finishing activity com.android.inputmethod.latin/com.google.android.apps.inputmethod.latin.preference.SettingsActivity
```

There are other people experiencing the same problem: [(1)](http://stackoverflow.com/q/41111670/1429387), [(2)](https://forum.xda-developers.com/showpost.php?p=70097116&postcount=1456), [(3)](https://forum.xda-developers.com/showpost.php?p=70103955&postcount=183). I'm using another keyboard as a workaround but I would like to know what's the cause of this problem.

**Comment by mfonville** [comment]
@naXa777 did you try it with a fully cleared `/data/` partition? There are known upgrade issues from the old Keyboard to the new GBoard.

**Comment by naXa777** [comment]
Resolved by reinstalling GBoard as a system app via `adb`.

**Comment by sd3993** [comment]
Can you please outline the steps? I downgraded to AOSP keyboard for now

**Comment by malt3** [comment]
*bump* Same problem on lineage OS.

**Comment by jamuir** [comment]
We build aosp images for several nougat devices (shamu, bullhead, angler, marlin, sailfish) and Gboard preferences seem to work fine.  Note that shamu is 32-bit, like hammerhead, which is the device mentioned by the OP.

It seems like this is only an issue when installing opengapps via sideload.

If so, then I don't know if we can address it by making changes inside aosp_build.

**Comment by paravz** [comment]
@naXa777 another bump: can you share how you reinstalled gboard to fix this

**Comment by hickscorp** [comment]
:+1

**Comment by sihaiyi** [comment]
Yeah, same issue for me ,could you please show the details of solution?

**Comment by naXa777** [comment]
@hickscorp The app may be located at /system/app/LatinIMEGooglePrebuilt/LatinIMEGooglePrebuilt.apk

**Comment by naXa777** [comment]
While I believe that the issue was caused by old settings of the previous keyboard, and installing the GBoard with a fully cleared /data partition should work, I personally used these commands to find all references to the GBoard app.

```
$ adb shell ls -lR | grep inputmethod
$ adb shell ls -lR | grep LatinIMEGooglePrebuilt
```

Then I've manually deleted them. Be careful cause you can delete something important accidentally.

Then I've replaced /system/app/LatinIMEGooglePrebuilt/LatinIMEGooglePrebuilt.apk file via `adb shell cp ...` (requires root permission) and executed `adb install -r GBoard.apk` in order to run installation process that properly unpacks some native libraries from the apk.

**Comment by paravz** [comment]
I can confirm that installing "clean" (with Data wipe) gapps with googlekeyboard works fine, no crashes in gboard settings. But all my attempts to fix a "dirty" installation failed - I reinstalled with a wipe.

**Comment by sihaiyi** [comment]
Hi,

Thanks for your response.

Yeah I wiped all data of my phone and re-install it and it seems to work
fine, it might be a conflict between AOSP and Gboard.
Problem solved, thanks a lot.

2017-03-13 20:10 GMT+01:00 Alec Istomin <notifications@github.com>:

**Comment by jamuir** [comment]
So this does indeed seem to be an issue caused by GBoard reading old data, which is what mfonville originally suggested :-)

I think this can be closed.

**Comment by paravz** [comment]
would be great "dirtiness" can be handled for googlekeyboard, it might be as easy as cleaning some files or prohibiting dirty flashing (as is done for googlecamera)

**Comment by jamuir** [comment]
agreed, but "aosp_build" is probably not the right place to fix that.

That sounds like something that needs to be done in the project that creates the opengapps zips.

**Comment by MattMski** [comment]
@naXa777 Can you explain in detail how you did that? Cause I've been deleting files and it's not working. I don't want to wipe all of my data either. I'd appreciate it, thanks!

**Comment by patricksylau** [comment]
Does anyone have detailed instructions on how this can be fixed? Thanks so much!

**Comment by sihaiyi** [comment]
Mine is just clean uninstall it or re-flash the whole rom

**Comment by dmitrygorbenko** [comment]
I've cleared Data partition and installed gapps again, and my issue was fixed.

**Comment by Fabian42** [comment]
Is there a way to fix this without resetting everything? On my Fairphone 2 I installed the "Fairphone open" system, updated it to Android 7.1, installed GBoard and the error was already there. There can be no old GBoard data anywhere, since I installed it after the update. So another reset shouldn't do anything useful. And I also don't **want** to reset.
Apart from that, it's still a bug in OpenGApps and should be fixed, so this report should be reopened.
